### PR TITLE
file structure 

### PR DESCRIPTION
**Pull request by momenmarwan**
make a file structure and setup environment and eslint and  typescript config

### CODE REVIEW

**Review by Farah202324 — APPROVED** [review]

**Review by momenqudaih — DISMISSED** [review]
Good job Momen

File: .github/workflows/github-actions-demo.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        run: npm run test
+
+        # Environments variables.
+
+        env:
+
+          DATABASE_TEST: ${{ secrets.TEST_DB }}
+
+          SECRET_KEY: ${{ secrets.SECRET_KEY }}
```
Comment (moustf) [inline .github/workflows/github-actions-demo.yml:82]: Remove the unnecessary spaces.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "editor.formatOnSave": false
+    },
+    "editor.codeActionsOnSave": {
+        "source.fixAll.eslint": true
+    },
+    "eslint.workingDirectories": [
+        "client"
+    ],
+    "editor.tabSize": 2
+}
```
Comment (moustf) [inline .vscode/settings.json:18]: Add `"files.eol": "\n"` to the settings.

File: client/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      "error",
+      {
+        "tabWidth": 2,
+        "printWidth": 80,
+        "useTabs": false,
+        "semi": true,
+        "singleQuote": true,
+        "trailingComma": "all",
+        "arrowParens": "avoid",
+        "endOfLine": "lf"
```
Comment (moustf) [inline client/.eslintrc.json:29]: Set the endOfLine's value to "auto" to solve the error of eol between windows and linux.

File: client/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable react/button-has-type */
+import react, { useState } from 'react';
+import reactLogo from './assets/react.svg';
+import './App.css';
+
+function App(): react.ReactElement {
```
Comment (moustf) [inline client/src/App.tsx:6]: Clear the component, remove the type, and convert it to an arrow function.
Use named exports.

File: client/src/App.css [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#root {
```
Comment (moustf) [inline client/src/App.css:1]: Remove the styles, no need for them.

File: server/__test__/test.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable no-undef */
```
Comment (moustf) [inline server/__test__/test.js:1]: When we use JavaScript files to test our routes, we won't be able use ES6 features like imports/exports, you need to modify it and use the previous configs.

File: server/app.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import express, { Express, json, urlencoded } from 'express';
+
+import cookieParser from 'cookie-parser';
+import compression from 'compression';
+
+import { router } from './routes/index';
+
+const app: Express = express();
```
Comment (moustf) [inline server/app.ts:8]: Remove the type of the variable, and use named exports.
```ts
export const app = express();
```

File: server/app.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+app.use([
+  json(),
+  urlencoded({ extended: false }),
+  compression(),
+  cookieParser(),
+]);
```
Comment (moustf) [inline server/app.ts:15]: Use cors middleware to allow other origins to reach to our server. http://localhost:5173 is seend as other origin.

File: server/routes/index.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { Router, Request, Response } from 'express';
+
+export const router: Router = Router();
```
Comment (moustf) [inline server/routes/index.ts:3]: Add a comment that this will be removed, and remove the type of the router.

**Review by Farah202324 — APPROVED** [review]

**Review by momenqudaih — APPROVED** [review]

**Review by moustf — APPROVED** [review]
GJ.

**Pull request merged**